app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java

```diff
@@ -456,31 +462,69 @@ public void onItemClick(int position) {
 
 
     /**
-     * Saves Current Image with grayscale filter
+     * Saves Images with grayscale filter
      */
-    private void saveCurrentImageInGrayscale() {
+    private void saveImagesInGrayscale() {
+        tempImageUri = new ArrayList<>();
         try {
             File sdCard = Environment.getExternalStorageDirectory();
             File dir = new File(sdCard.getAbsolutePath() + "/PDFfilter");
             dir.mkdirs();
-            Picasso picasso = Picasso.with(mActivity);
-            Transformation transformation = new GrayscaleTransformation();
 
-            for (int countElements = mImagesUri.size() - 1; countElements >= 0; countElements--) {
+            int size = mImagesUri.size();
+            for (int i = 0; i < size; i++) {
                 String fileName = String.format(getString(R.string.filter_file_name),
-                        String.valueOf(System.currentTimeMillis()), "grayscale");
+                        String.valueOf(System.currentTimeMillis()), i + "_grayscale");
                 File outFile = new File(dir, fileName);
                 String imagePath = outFile.getAbsolutePath();
-                picasso.load(new File(mImagesUri.get(countElements)))
-                        .transform(transformation)
-                        .into(getTarget(imagePath));
-                mImagesUri.remove(countElements);
+
+                File f = new File(mImagesUri.get(i));
+                FileInputStream fis = new FileInputStream(f);
+                Bitmap bitmap = BitmapFactory.decodeStream(fis);
+                Bitmap grayScaleBitmap = toGrayscale(bitmap);
+
+                File file = new File(imagePath);
+                file.createNewFile();
+                FileOutputStream ostream = new FileOutputStream(file);
+                BufferedOutputStream bos = new BufferedOutputStream(ostream, 1024 * 8);
+                grayScaleBitmap.compress(Bitmap.CompressFormat.JPEG, 100, bos);
+                bos.flush();
+                bos.close();
+                ostream.close();
+                tempImageUri.add(imagePath);
             }
+            mImagesUri.clear();
+            mImagesUri.addAll(tempImageUri);
         } catch (SecurityException e) {
             e.printStackTrace();
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
         }
     }
 
+    /**
+     * convert a bitmap to grayscale and return it
+     * @param bmpOriginal original bitmap which is converted to a new
+     *                    grayscale bitmap
+     */
+    private Bitmap toGrayscale(Bitmap bmpOriginal) {
```

Comment:
you can add this function to imageutils file

Reply:
review the changes again